src/Platforms/OMPTarget/ompBLAS.cpp

```diff
@@ -411,5 +442,150 @@ ompBLAS_status ger_batched(ompBLAS_handle&                   handle,
+                            float* const       y[],
+                            const int          incy,
+                            const int          batch_count)
+{
+  return copy_batched_impl(handle, n, x, incx, y, incy, batch_count);
+}
+
+ompBLAS_status copy_batched(ompBLAS_handle&     handle,
+                            const int           n,
+                            const double* const x[],
+                            const int           incx,
+                            double* const       y[],
+                            const int           incy,
+                            const int           batch_count)
+{
+  return copy_batched_impl(handle, n, x, incx, y, incy, batch_count);
+}
+
+#if !defined(OPENMP_NO_COMPLEX)
+ompBLAS_status copy_batched(ompBLAS_handle&                  handle,
+                            const int                        n,
+                            const std::complex<float>* const x[],
+                            const int                        incx,
+                            std::complex<float>* const       y[],
+                            const int                        incy,
+                            const int                        batch_count)
+{
+  return copy_batched_impl(handle, n, x, incx, y, incy, batch_count);
+}
+
+ompBLAS_status copy_batched(ompBLAS_handle&                   handle,
+                            const int                         n,
+                            const std::complex<double>* const x[],
+                            const int                         incx,
+                            std::complex<double>* const       y[],
+                            const int                         incy,
+                            const int                         batch_count)
+{
+  return copy_batched_impl(handle, n, x, incx, y, incy, batch_count);
+}
+#endif
+
+template<typename T>
+ompBLAS_status copy_batched_offset_impl(ompBLAS_handle& handle,
+                                        const int       n,
+                                        const T* const  x[],
+                                        const int       x_offset,
+                                        const int       incx,
+                                        T* const        y[],
+                                        const int       y_offset,
+                                        const int       incy,
+                                        const int       batch_count)
+{
+  if (batch_count == 0) return 0;
+
+  //if (incx !=1 || incy != 1)
+  //  throw std::runtime_error("incx !=1 or incy != 1 are not implemented in ompBLAS::copy_batched_impl!");
+
+  PRAGMA_OFFLOAD("omp target teams distribute parallel for collapse(2) is_device_ptr(x, y)")
+  for (size_t ib = 0; ib < batch_count; ib++)
+    for (size_t i = 0; i < n; i++)
+      y[ib][y_offset + i * incy] = x[ib][x_offset + i * incx];
+  return 0;
+}
+
+ompBLAS_status copy_batched_offset(ompBLAS_handle&    handle,
```

Comment:
This is an extension of BLAS. So please document this function especially its arguments.

Reply:
I added some documentation in ompBLAS.hpp